### PR TITLE
feat: add support for StatefulSets

### PR DESCRIPTION
**Pull request by Soumil-07**

### CODE REVIEW

File: types/qaengine/commonqa/commonqa.go [hunk 1]
```diff
@@ -151,3 +151,9 @@ func GetPortForService(detectedPorts []int32, qaSubKey string) int32 {
 	}
 	return int32(selectedPort)
 }
+
+// Stateful returns whether the Service should generate a StatefulSet
+func Stateful(serviceName string) bool {
```
[the remark this answers is not in the document]
Comment (Soumil-07) [reply types/qaengine/commonqa/commonqa.go]: @Akash-Nayak I've made the requested changes in 3db3bae

**Review by Akash-Nayak — COMMENTED** [review]
I was doing some testing and noticed that in the generated target artifacts, Move2Kube also created Route/Ingress yaml for the services for which I selected yes for StatefuSet and ClusterIP as the service type. I think Ingress/Route should only be created for the service if the user selects Ingress as the service type during the QA.
Can you please also check that and look into this issue @Soumil-07.
cc: @HarikrishnanBalagopal

**Comment by Soumil-07** [comment]
> I was doing some testing and noticed that in the generated target artifacts, Move2Kube also created Route/Ingress yaml for the services for which I selected yes for StatefuSet and ClusterIP as the service type. I think Ingress/Route should only be created for the service if the user selects Ingress as the service type during the QA.
> Can you please also check that and look into this issue @Soumil-07.
> cc: @HarikrishnanBalagopal

It should be resolved by 50c367b.

**Review by Akash-Nayak — APPROVED** [review]
LGTM. Thanks @Soumil-07!

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**